### DISCUSSION TITLE
Toggle automatic screen switching in osu tournament client

### DISCUSSION THREAD

**Discussion by rodrigopina360** [ppy/osu#20038] [Ideas] [answered]
Currently, whenever a map is played, after ther results screen, the tournament client will automatically change to either the mappool screen, or the team victory screen, and close enough behaviour happens in the mappool screen as well. Although this is very useful for must people, it can be annoying during live streams with transitions between screens, as you might be comentating about something in the results screen and it automatically changes in the middle of your talk. 
My personal suggestion after trying to look into the code (without much success) would be to surround the code responsible for changing scheduledOperation's value in GameplayScreen and MapPoolScreen in an if statement, where it would only run if a toggle on SetupScreen was true.
I tried implementing a fix, but without much knowledge on how data is moved between screens, I was unsuccessful in linking a bool variable to a checkbox, and reading it on other screens.
![1](https://user-images.githubusercontent.com/41023844/187557969-99226055-c21d-423b-a096-58a0cc668380.png)
![2](https://user-images.githubusercontent.com/41023844/187557971-f72c3446-0564-48a6-9dcd-0b3b2739ba12.png)

**Comment by peppy** [comment, marked as answer]
I'll take a look into this.

**Reply by peppy** [reply]
@Pinossaur check out https://github.com/ppy/osu/pull/20041

the flow is pretty simple once you understand where things sit!

**Reply by rodrigopina360** [reply]
Thank you for the quick answer, I was just having a hard time finding a way to have the value of the toggle button have an effect on the other screens, still a pretty junior developer, but it was really fun to dig in and (try to) find ways to fix the issue.
